dev/tests/integration/framework/Magento/TestFramework/Workaround/IsolationLogger.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\TestFramework\Workaround;
+
+
+
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\App\ResourceConnection;
+
+/**
+ * Trigger queue to process storefront consumers
+ */
+class IsolationLogger
+{
+    private $previousTest = 'N/A';
+    private $currenTest;
+
+    /**
+     * Handler for 'startTest' event.
+     *
+     * Sync Magento monolith App data with Catalog Storefront Storage.
+     *
+     * @param \PHPUnit\Framework\TestCase $test
+     * @return void
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function startTest(\PHPUnit\Framework\TestCase $test)
+    {
+        $this->currenTest = \get_class($test) . '::' .  $test->getName(true);
+        foreach ($this->dbStateTables as $dbStateTable => $method) {
+            $data = $this->pullDbState($dbStateTable);
+            $this->{$method}($data);
+        }
+
+        // set current test as "previous"
+        $this->previousTest = \get_class($test) . '::' .  $test->getName(true);
+
+    }
+
+    /**
+     * @var string[]
+     */
+    private $dbStateTables = [
+        'catalog_product_entity' => 'assertIsEmpty',
+        'eav_attribute' => 'eavAttributeAssert',
+        'catalog_category_entity' => 'assertTwoRecords',
+        'eav_attribute_set' => 'attributeSetAssert',
+        'store' => 'assertTwoRecords',
+        //'catalog_data_exporter_products' => 'assertIsEmpty',
+        //'catalog_data_exporter_product_attributes' => 'assertIsEmpty',
+        //'catalog_data_exporter_categories' => 'assertIsEmpty'
```

Comment:
I think we can remove these lines, or uncomment them

Reply:
It's just an example and will no go to mainline. For mainline we need to have another approach. task: https://github.com/magento/catalog-storefront/issues/210
Fill free to join! )

Reply:
Unfortunately I don’t have access